frontend/src/components/pages/UserProfilePage.tsx

```diff
@@ -1,18 +1,162 @@
-import React from "react";
-import { useParams } from "react-router-dom";
-import { Box } from "@chakra-ui/react";
+import React, { useState } from "react";
+import { useParams, useHistory } from "react-router-dom";
+import {
+  Box,
+  Button,
+  Flex,
+  Heading,
+  Text,
+  useStyleConfig,
+} from "@chakra-ui/react";
+import { useQuery, useMutation } from "@apollo/client";
+import Header from "../navigation/Header";
+import {
+  SoftDeleteUserResponse,
+  SOFT_DELETE_USER,
+} from "../../APIClients/mutations/UserMutations";
+import {
+  MANAGE_USERS_TABLE_DELETE_USER_BUTTON,
+  MANAGE_USERS_TABLE_DELETE_USER_CONFIRMATION,
+} from "../../utils/Copy";
+import ConfirmationModal from "../utils/ConfirmationModal";
+import { GET_USER, User } from "../../APIClients/queries/UserQueries";
 
 type UserProfilePageProps = {
-  userId: string | undefined;
+  userId: string;
 };
 
 const UserProfilePage = () => {
   const { userId } = useParams<UserProfilePageProps>();
 
+  const [fullName, setFullName] = useState<string>("");
+  const [email, setEmail] = useState<string>("");
+  const [role, setRole] = useState<string>("");
+  const [confirmDeleteUser, setConfirmDeleteUser] = useState(false);
+
+  /* eslint-disable */
+  const [approvedLanguagesTranslation, setApprovedLanguagesTranslation] =
+    useState<{ [name: string]: number }>();
```

Comment:
just curious, why do you use `name` here? is `name` a necessary key or would the following also work?
`useState<{ string: number }>();`
`useState<{ [language: string]: [level: number] }>();`

Reply:
oh I got it from [here](https://github.com/uwblueprint/planet-read/blob/main/frontend/src/components/homepage/Filter.tsx#L15), i can change it to `useState<{ [language: string]: number}>();` tho